x/dex/keeper/msg_server_place_orders.go

```diff
@@ -44,6 +45,20 @@ func (k msgServer) PlaceOrders(goCtx context.Context, msg *types.MsgPlaceOrders)
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
+	calculatedCollateral := sdk.NewDecFromBigInt(big.NewInt(0))
+	for _, order := range msg.Orders {
+		calculatedCollateral = calculatedCollateral.Add(order.Price.Mul(order.Quantity))
+	}
+
+	if msg.AutoCalculateDeposit {
+		// throw error if current funds amount is less than calculatedCollateral
+		if msg.Funds[0].Amount.LT(calculatedCollateral.Mul(sdk.NewDecFromInt(sdk.NewInt(1000000))).RoundInt()) {
```

Comment:
I probably didn't describe this clearly enough, but we don't really care what's passed in `msg.Funds` by the user. As long as `msg. AutoCalculateDeposit` is true, and there are enough funds in the user's bank, we will execute line 58-59. To check user's bank balance you can do `k.Keeper.BankKeeper.GetBalance(ctx sdk.Context, addr sdk.AccAddress, denom string)`